mesonbuild/mconf.py

```diff
@@ -299,22 +299,23 @@ def print_nondefault_buildtype_options(self):
 def run(options):
     coredata.parse_cmd_line_options(options)
     builddir = os.path.abspath(os.path.realpath(options.builddir))
+    print_only = not options.cmd_line_options and not options.clearcache
     c = None
     try:
         c = Conf(builddir)
-        if c.default_values_only:
+        if c.default_values_only and not print_only:
+            raise MesonException('No valid build directory found, cannot modify options.')
```

Comment:
So this needs to change then:

```
WARNING: The source directory instead of the build directory was specified.
WARNING: Only the default values for the project are printed, and all command line parameters are ignored.
```

"and all command line parameters are ignored" is no longer true, which technically breaks documented UX. Although I indeed can't imagine why someone would be doing this.

Reply:
fixed.

Reply:
@dcbaker, what are your thoughts on the changed UX?